drivers/gnss/gnss_seeed_air530z.c

```diff
@@ -0,0 +1,334 @@
+}
+
+static int gnss_seeed_air530z_init_chat(const struct device *dev)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+
+	const struct modem_chat_config chat_config = {
+		.user_data = data,
+		.receive_buf = data->chat_receive_buf,
+		.receive_buf_size = sizeof(data->chat_receive_buf),
+		.delimiter = data->chat_delimiter,
+		.delimiter_size = ARRAY_SIZE(data->chat_delimiter),
+		.filter = NULL,
+		.filter_size = 0,
+		.argv = data->chat_argv,
+		.argv_size = ARRAY_SIZE(data->chat_argv),
+		.unsol_matches = unsol_matches,
+		.unsol_matches_size = ARRAY_SIZE(unsol_matches),
+	};
+
+	return modem_chat_init(&data->chat, &chat_config);
+}
+
+static void seeed_air530z_init_dynamic_script(const struct device *dev)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+
+	/* Air530z doesn't respond to commands. Thus, response_matches_size = 0; */
+	data->dynamic_script_chat.request = data->dynamic_request_buf;
+	data->dynamic_script_chat.response_matches = NULL;
+	data->dynamic_script_chat.response_matches_size = 0;
+	data->dynamic_script_chat.timeout = 0;
+
+	data->dynamic_script.name = "PCAS";
+	data->dynamic_script.script_chats = &data->dynamic_script_chat;
+	data->dynamic_script.script_chats_size = 1;
+	data->dynamic_script.abort_matches = NULL;
+	data->dynamic_script.abort_matches_size = 0;
+	data->dynamic_script.callback = NULL;
+	data->dynamic_script.timeout = 5;
+}
+
+static int gnss_seeed_air530z_init(const struct device *dev)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+	const struct gnss_seeed_air530z_config *config = dev->config;
+	int ret;
+
+	k_sem_init(&data->lock, 1, 1);
+
+	ret = gnss_seeed_air530z_init_nmea0183_match(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	gnss_seeed_air530z_init_pipe(dev);
+
+	ret = gnss_seeed_air530z_init_chat(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	seeed_air530z_init_dynamic_script(dev);
+
+	ret = modem_pipe_open(data->uart_pipe);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = modem_chat_attach(&data->chat, data->uart_pipe);
+	if (ret < 0) {
+		modem_pipe_close(data->uart_pipe);
+		return ret;
+	}
+
+	ret = modem_chat_run_script(&data->chat, &init_script);
+	if (ret < 0) {
+		LOG_ERR("Failed to run init_script");
+		modem_pipe_close(data->uart_pipe);
+		return ret;
+	}
+
+	/* setup on-off gpio for power management */
+	if (!gpio_is_ready_dt(&config->on_off_gpio)) {
+		LOG_ERR("on-off GPIO device not ready");
+		return -ENODEV;
+	}
+
+	gpio_pin_configure_dt(&config->on_off_gpio, GPIO_OUTPUT_LOW);
+
+	return 0;
+}
+
+static int seeed_air530z_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	struct gnss_seeed_air530z_data *data = dev->data;
+	const struct gnss_seeed_air530z_config *config = dev->config;
+	int ret = -ENOTSUP;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_SUSPEND:
+		gpio_pin_set_dt(&config->on_off_gpio, 0);
+		ret = 0;
+		break;
+
+	case PM_DEVICE_ACTION_RESUME:
+		gpio_pin_set_dt(&config->on_off_gpio, 1);
+		ret = 0;
+		break;
```

Comment:
When suspending, you should close the modem_pipe (which disables the UART RX) :) The UART may generate RX errors if the GNSS modem turns off the UART transciever (which it most likely does), it is also a waste of power to keep it enabled :)

Reply:
Hi @bjarki-trackunit thank you very much for the feedback, I am learning so much with this PR :)

I have added the requested pipe closing and openning code during Power Management actions. The power saving is not so significant (I have mesaured it). In any case, it is true that is better to release the UART if it is not using.

The driver is performing great :D